R/tm_g_km.R

```diff
@@ -656,18 +655,67 @@ srv_g_km <- function(id,
         facet_var = facet_var,
         time_unit_var = time_unit_var
       ),
-      merge_function = "dplyr::inner_join",
-      join_keys = get_join_keys(data)
+      datasets = data,
+      select_validation_rule = list(
+        aval_var = shinyvalidate::sv_required("An analysis variable is required"),
+        cnsr_var = shinyvalidate::sv_required("A censor variable is required"),
+        arm_var = shinyvalidate::sv_required("A treatment variable is required")
+      ),
+      filter_validation_rule = list(
+        paramcd = shinyvalidate::sv_required("An endpoint is required")
+      )
     )
 
-    anl_q <- reactive({
+    iv_r <- shiny::reactive({
+      iv <- shinyvalidate::InputValidator$new()
+
+      if (isTRUE(input$compare_arms)) {
+        iv$add_validator(iv_arm_ref)
+      }
+
+      iv$add_rule("font_size", shinyvalidate::sv_required("Plot tables font size must be greater than or equal to 5"))
+      iv$add_rule("font_size", shinyvalidate::sv_gte(5, "Plot tables font size must be greater than or equal to 5"))
+      iv$add_rule("conf_level", shinyvalidate::sv_required("Please choose a confidence level"))
+      iv$add_rule(
+        "conf_level",
+        shinyvalidate::sv_between(
+          0, 1,
+          inclusive = c(FALSE, FALSE),
+          message_fmt = "Confidence level must be between 0 and 1"
+        )
+      )
+      iv$add_rule("xticks", shinyvalidate::sv_optional())
+      iv$add_rule(
+        "xticks",
+        function(value) {
+          val <- as_numeric_from_comma_sep_str(value, sep = ";")
+          if (anyNA(val) || any(val < 0)) {
+            "All break intervals for x-axis must be non-negative numbers"
```

Comment:
Previously user could provide numbers separated by semicolon and comma. Having NA here means that someone provided wrong number (not convertible to numeric directly)
```suggestion
            "All break intervals for x-axis must be non-negative numbers separated by semicolon"
```